app/services/degrees/dfe_reference.rb

```diff
@@ -25,12 +25,25 @@ def find_grade(uuid: nil, name: nil, hesa_code: nil)
       end
 
       def find_item(list_type, filters)
-        DfE::ReferenceData::Degrees.const_get(list_type.to_s.upcase).all.find do |record|
+        refdataset = DfE::ReferenceData::Degrees.const_get(list_type.to_s.upcase)
+
+        refdataset.all.find do |record|
           filters.compact.any? do |field, value|
-            same_string?(record[field], value)
+            if field.to_s.include?("name") || field.to_s.include?("abbreviation")
```

Comment:
```suggestion
            if field == :name || field == :abbreviation
```

Reply:
nice, I originally had what you've suggested but it made rubocop complain! I'll put it back as you suggested